gen/mailman_elixir

```diff
@@ -36,27 +36,28 @@ mkdir $mailinglistsDirectory or die "Can't mkdir $mailinglistsDirectory: $!";
 
 
 my @membersAttributes;
-my @resourcesData = $data->getChildElements;
-foreach my $rData (@resourcesData) {
-    my %resourceAttributes  = attributesToHash $rData->getAttributes;
+foreach my $resourceId ($data->getResourceIds()) {
 
-    my @membersData = $rData->getChildElements;
+	my $listName = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_NAME);
+	my $adminMail = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_MAILING_LIST_MANAGER_MAIL);
 
-    my $listName = $resourceAttributes{$A_RESOURCE_MAILING_LIST_NAME};
-    my $adminMail = $resourceAttributes{$A_RESOURCE_MAILING_LIST_MANAGER_MAIL};
+	# Store the admin's mail
+	$mailinglistAdminMailStruc->{$listName} = $adminMail;
 
-    # Store the admin's mail
-    $mailinglistAdminMailStruc->{$listName} = $adminMail;
+	# Init mailinglistStruct with the listName
+	$mailinglistStruc->{$listName} = {};
 
-    # Init mailinglistStruct with the listName
-    $mailinglistStruc->{$listName} = {};
+	foreach my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
 
-    foreach my $memberAttributes (dataToAttributesHashes @membersData) {
-        #list only VALID members except if allowNonvalidUsers is true
-        next unless($memberAttributes->{$A_USER_STATUS} eq "VALID");
+		my $memberStatus = $data->getMemberAttributeValue(member => $memberId, attrName => $A_USER_STATUS);
+		my $userMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_MAIL);
+		my $userName = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_NAME);
 
-        $mailinglistStruc->{$listName}->{$memberAttributes->{$A_USER_MAIL}}->{$A_USER_NAME} = $memberAttributes->{$A_USER_NAME};
-    }
+		#list only VALID members except if allowNonvalidUsers is true
```

Comment:
I don't think, that we process ``allowNonvalidUsers`` anywhere in the script, so the comment is misleading.

Reply:
Fixed